### PR TITLE
fix(publish) - Nested packages were overwritten by the parent in publ…

### PR DESCRIPTION
**Pull request by akintolga**
…ishing

### CODE REVIEW

File: server/apps/publish/archive_publish.py [hunk 1]
```diff
@@ -245,8 +245,11 @@ def _publish_package_items(self, package, updates):
 
                     if package_item[ITEM_TYPE] == CONTENT_TYPE.COMPOSITE:
                         # if the item is a package do recursion to publish
+                        wanted_keys = ['state', 'operation']
+                        sub_updates = dict([(i, updates[i]) for i in wanted_keys if i in updates])
```
Comment (plamut) [inline server/apps/publish/archive_publish.py]: Creating a temporary list is not really needed, the same can be done more efficiently with a dict comprehension
```py
sub_updates = {i: updates[i] for i in wanted_keys if i in updates}
```

**Comment by plamut** [comment]
Where are the tests? :)

**Comment by ioanpocol** [comment]
:+1:  (except already mentioned things)

**Comment by mugurrus** [comment]
:+1: (I don't think new tests are needed, you can update existing tests)

**Comment by sivakuna-aap** [comment]
:+1:

**Comment by ioanpocol** [comment]
Looks that this PR fixes one behave test that is failing. I will merge it and any other changes will be done in other PR.

**Pull request merged**